internal/scorer/algorithm/wam/wam_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2022 Criticality Score Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wam
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/ossf/criticality_score/internal/scorer/algorithm"
+)
+
+func TestWeighetedArithmeticMean_Score(t *testing.T) {
+	//nolint:govet
+	tests := []struct {
+		name   string
+		inputs []*algorithm.Input
+		record map[string]float64
+		want   float64
+		err    error
+	}{
+		{
+			name: "regular test",
+			inputs: []*algorithm.Input{
+				{
+					Weight: 1, Distribution: algorithm.LookupDistribution("linear"),
+					Source: algorithm.Value(algorithm.Field("1")),
+				},
+				{
+					Weight: 2, Distribution: algorithm.LookupDistribution("linear"),
+					Source: algorithm.Value(algorithm.Field("2")),
+				},
+				{
+					Weight: 3, Distribution: algorithm.LookupDistribution("linear"),
+					Source: algorithm.Value(algorithm.Field("3")),
+				},
+				{
+					Weight: 4, Distribution: algorithm.LookupDistribution("linear"),
+					Source: algorithm.Value(algorithm.Field("4")),
+				},
+				{
+					Weight: 5, Distribution: algorithm.LookupDistribution("linear"),
+					Source: algorithm.Value(algorithm.Field("5")),
+				},
+			},
+			want:   3.6666666666666665,
+			record: map[string]float64{"1": 1, "2": 2, "3": 3, "4": 4, "5": 5},
+		},
+		{
```

Comment:
If these are blocked when the Input is created then I don't think these "negative weight" or "zero weight" tests need to be here.

However there should be tests here for edge cases, for example:

- no fields matching the record
- partial matches between the defined fields and the record

Reply:
I have removed the negative weight and zero weight tests, and have put in the tests `"fields not matching the record"` and `"some fields matching the record"`.